scipy/stats/_continuous_distns.py

```diff
@@ -8806,6 +8806,138 @@ def _entropy(self, kappa):
         return (-kappa * sc.i1(kappa) / sc.i0(kappa) +
                 np.log(2 * np.pi * sc.i0(kappa)))
 
+    def expect(self, func=None, args=(), loc=0, scale=1, lb=-np.pi, ub=np.pi,
+                conditional=False, **kwds):
+        """Calculate expected value of a function with respect to the
+        distribution by numerical integration.
+
+        The expected value of a function ``f(x)`` with respect to a
+        distribution ``dist`` is defined as::
+
+                    ub
+            E[f(x)] = Integral(f(x) * dist.pdf(x)),
+                    lb
+
+        where ``ub`` and ``lb`` are arguments and ``x`` has the ``dist.pdf(x)``
+        distribution. If the bounds ``lb`` and ``ub`` correspond to the
+        support of the distribution, e.g. ``[-pi, pi]`` (or any interval
+        spanning ``2pi``) in the default
+        case, then the integral is the unrestricted expectation of ``f(x)``.
+        Also, the function ``f(x)`` may be defined such that ``f(x)`` is ``0``
+        outside a finite interval in which case the expectation is
+        calculated within the finite range ``[lb, ub]``.
+
+        Parameters
+        ----------
+        func : callable, optional
+            Function for which integral is calculated. Takes only one argument,
+            and can return a complex number.
+            The default (`np.exp(1j*x)`) results in the mean direction.
+        args : tuple, optional
+            Shape parameters of the distribution.
+        loc : float, optional
+            Location parameter (default=0).
+        scale : float, optional
+            Scale parameter (default=1).
+        lb, ub : scalar, optional
+            Lower and upper bound for integration. Default is set [-np.pi,np.pi]
+        conditional : bool, optional
+            If True, the integral is corrected by the conditional probability
+            of the integration interval.  The return value is the expectation
+            of the function, conditional on being in the given interval.
+            Default is False.
+
+        Additional keyword arguments are passed to the integration routine.
+
+        Returns
+        -------
+        expect : float
+            The calculated expected value.
+
+        Notes
+        -----
+        The integration behavior of this function is inherited from
+        `scipy.integrate.quad`. Neither this function nor
+        `scipy.integrate.quad` can verify whether the integral exists or is
+        finite. For example ``cauchy(0).mean()`` returns ``np.nan`` and
+        ``cauchy(0).expect()`` returns ``0.0``.  It is often the case in
+        directional statistics to supply complex expontial functions
```

Comment:
I had not initially considered this.  I haven't thought how general the solution is to integration.  Finding a general purpose routine could be much more complex for all of the possible uses.

Reply:
> I haven't thought how general the solution is to integration.

I don't understand. It seems that applying this approach to `quad` itself is no more or less general than it is here inside the `expect` method. In both cases, the user specifies an integrand function, a lower limit, and an upper limit. The only difference in `expect` is that the user's integrand gets multiplied by a PDF. In either case, this approach is only going to be accurate if the function is analytic, otherwise, the user should parametrize the path to be taken with a real variable.

So I'm suggesting that inside `quad`, you take the same approach as you do here - if the integral is complex, use `quad` on the real and complex parts separately. To detect whether the integral is complex, you can check whether either of the limits are complex. Even if they are both real, the integrand could still be complex, which you can detect by evaluating the integrand at one of the integration limits and checking the output `dtype`.

Maybe I'm wrong, though. Let me know.

Reply:
You are right.  Sorry, I didn't mean to imply it's not a general method to integrate a complex function.  I was really speculating on general efficiency and design acceptance.  For instance, could this solution in `quad` get in the way of users of quad to solve certain problems in a more efficient way? Another concern is older that may use `quad` in the same manner as this PR to integrate a complex integrand.  Such change would introduce a redundant integration - it would compute the real part of the purely complex component of the integrand, and it would sneak in this inefficiency.. 

Finally, it seems like detecting a complex integrand could be misleading.  $e^{i\theta} =\ cos(\theta) + i \sin(\theta)$ should be real at $\theta = n\pi$ for all integer values of $n$.  Only by some numeric artifact does `np.iscomplex` return `True` here.  

I cam across [this](https://mail.python.org/archives/list/scipy-dev@python.org/message/OZOZTYXYFLGE7QT76CBWJYTE46GKXFMS/) discussion on a somewhat related matter. There are a couple of points worth considering there too.  Most importantly, it seems like a design stance SciPy has taken that users must opt into the complex domain.  

I take your point.   I'm happy to take a swing at it.  However, I do think there is a good chance we will revert back to this solution.  Do you think it would better to open a new PR to update `quad` (then merge into this branch)?

Reply:
Those are ways it *could* be detected automatically, but it doesn't have to be automatic. We could add a keyword if need be.

Yes, I think it would be better to open a PR against `quad`. We can change this PR now to just fix the default integration limits but stick to real integrals for now. Then, if `quad` gets the ability to do complex integrals, we can make additional changes if necessary to add that capability to `except` - for all distributions, not just von Mises.

Reply:
@cowden would you send a message to the mailing list about your PR? Maybe something like mine about [`goodness_of_fit`](https://mail.python.org/archives/list/scipy-dev@python.org/message/3BIOIOFKQZ55KVT7BX5P6WHXFVRRBCSX/) inviting input before we release with a new keyword parameter.

Reply:
I will send a note shortly.